tests/tpg300.py

```diff
@@ -68,9 +64,12 @@ def test_that_GIVEN_a_connected_emulator_WHEN_ioc_started_THEN_ioc_is_not_disabl
     @skip_if_recsim("Requires emulator")
     def test_that_GIVEN_a_connected_emulator_WHEN_units_are_set_THEN_unit_is_the_same_as_backdoor(self):
         for unit in Units:
-            self._set_units(unit)
-
             expected_unit = unit.name
+            self.ca.set_pv_value("UNITS:SP", expected_unit)
+            device_unit_val = int(self._lewis.backdoor_run_function_on_device("backdoor_get_unit")[0])
+
+            self.assertEqual(device_unit_val, unit.value)
```

Comment:
Is it possible to use an emulator assertion e.g. `self._lewis.assert_that_emulator_value_is` here? If not maybe we can add an assertion to the test framework for this?

Otherwise this is a bit of a race condition between the test and the IOC sending the new value to the emulator.